### PR TITLE
Add currentLocation feature in the 'Expanding Network' section

### PR DESCRIPTION
**Pull request by RushilGupta4**

### CODE REVIEW

File: source/firstprogram-expanding-network.rst [hunk 1]
```diff
@@ -181,159 +185,324 @@ With these values defined, ``create24HourSchedules`` can be made. However, when
 
 .. note:: The timings of departure and return are to be made in the 24 hour format.  
 
-The whole main file code is 
+
+Handling Current Locations
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Now, we also need to handle the location of the individual at every step to ensure that the individual is only in contact with the people in the same location. This is done by adding a new ``currentLocation`` parameter in the person class. We modify the constructor to include this parameter.
```
Comment (dpcherian) [inline source/firstprogram-expanding-network.rst:191]: Typeset class names with backticks and correct capitalisation: for example, 

> [...] in the ``Person`` class.

Comment (dpcherian) [reply source/firstprogram-expanding-network.rst:191]: We haven't used ideas like "constructors" yet, could you say something that would be a little more clear to novice users like:

> In order to implement this, we add a new attribute to the ``Person`` class called ``currentLocation`` which stores the exact current location of the agent. This attribute has to be dynamically updated at every time-step based on the agent's schedule. This is done by adding a new behaviour <add a local link here to the section on agents and behaviours> ``updateCurrentLocation``. This is implemented below.

File: source/firstprogram-expanding-network.rst [hunk 2]
```diff
@@ -181,159 +185,324 @@ With these values defined, ``create24HourSchedules`` can be made. However, when
...
-        logger.debug("Ingestion done")
-      })
+    private def getCurrentLocation(context: Context): Option[String] = {
+      val schedule = context.fetchScheduleFor(this).get
+      val currentStep = context.getCurrentStep
+      val placeType: String = schedule.getForStep(currentStep)
 
-      simulation.defineSimulation(implicit context => {
-        create24HourSchedules()
+      Some(placeType)
```
Comment (dpcherian) [inline source/firstprogram-expanding-network.rst:206]: Could you add one line below this explaining what ``Option`` and ``Some`` mean? Nothing too elaborate, just a small description like from the link you sent me (https://stackoverflow.com/questions/49944872/understanding-some-and-option-in-scala). Add a link to some online documentation on this as well, if possible.

File: source/firstprogram-expanding-network.rst [hunk 3]
```diff
@@ -181,159 +185,324 @@ With these values defined, ``create24HourSchedules`` can be made. However, when
...
           }
+        }
+
+        case _ => 
+      }
+    }
+
+    addBehaviour(updateCurrentLocation)
+
+We then also need to update the ``checkForInfection`` function to only check for infections in the same location. We start by counting the total number of people in the Person's current location, and then count the number of infected people in the same location. Then, we modify our infection rate calculation to be based on the number of infected people in the same location.
```
Comment (dpcherian) [inline source/firstprogram-expanding-network.rst:224]: Add one line here explaining what would go wrong if we didn't do this (everyone linked to a node would be considered, even if they were home).

**Review by dpcherian — CHANGES_REQUESTED** [review]
Good work! I've added a couple of simple changes that need to be made before merging, could you get them done? Thanks!

**Comment by RushilGupta4** [comment]
I have made the changes. Could you go over them once?

**Pull request merged**